### ISSUE TITLE
Migrate from Bintray

### ISSUE THREAD

**Issue by MrPowerGamerBR** [NuVotifier/NuVotifier#237]
[Bintray was sunsetted this month](https://jfrog.com/blog/into-the-sunset-bintray-jcenter-gocenter-and-chartcenter/), so currently there isn't a repository hosting NuVotifier, this makes it a pain to use NuVotifier as a dependency :(

Currently, as a alternative, [using Jitpack](https://jitpack.io/#NuVotifier/NuVotifier/2.7.2) seems to work fine, however the [Developer Documentation](https://github.com/NuVotifier/NuVotifier/wiki/Developer-Documentation) needs to be updated.

**Comment by Ichbinjoe** [comment]
Thats dope - I'll update the documentation. Thanks a ton for pointing this out!

**Comment by Ichbinjoe** [comment]
Updated! Thanks!

**Comment by HexedHero** [comment]
Sadly, This isn't a fix.

This isn't the same as the normal old Bintray dependency and doesn't contain the same things such as the events, for example, all you can get from this is the Vote object nothing else. The developer documentation also wasn't updated with the new dependency too.

**Comment by MrPowerGamerBR** [comment]
@HexedHero good catch! I forgot to update my original post to mention that you need to import both dependencies:

https://github.com/SparklyPower/SparklyPower/blob/master/bukkit/DreamVote/build.gradle.kts#L16

A proper solution is hosting a Maven repository, but for now, this works. :)

**Comment by Ichbinjoe** [comment]
Eh looks like jitpack doesn't do everything we like (#238). I have to make some infrastructure changes on my end to support this, so it'll be a few days.

**Comment by HexedHero** [comment]
> Eh looks like jitpack doesn't do everything we like (#238). I have to make some infrastructure changes on my end to support this, so it'll be a few days.

I suggested to CodeMC to add NuVotifier here https://discord.com/invite/AGcFMu6 but haven't gotten any reply, maybe if the developer goes to them it can be added and you'll never have to worry about the repo again.

**Comment by Ichbinjoe** [comment]
@HexedHero I appreciate you reaching out to them - I wasn't aware of this as a resource. I have pinged their channel additionally & will try to follow up in a few days. Otherwise. I have servers that _could_ be used for this but I need to upgrade some of the hardware on them before I host this myself.

**Comment by MrPowerGamerBR** [comment]
> 
> Eh looks like jitpack doesn't do everything we like (#238). I have to make some infrastructure changes on my end to support this, so it'll be a few days.

Just a FYI, I tried importing the BungeeCord dependency from Jitpack and didn't have any issues with it.

`compileOnly("com.github.NuVotifier.NuVotifier:nuvotifier-bungeecord:2.7.2")`

However I think that having a proper Maven repository is still a good idea.

**Comment by Lorenzo0111** [comment]
You should update the groupId on the wiki to `com.github.NuVotifier`

**Comment by itxfrosty** [comment]
Is there another way to get the universal package? Also the Wiki should be updated to use the new maven details.

**Comment by socketbat** [comment]
For anyone still having a problem using the maven dependency:
Try to use this older version from jitpack:

`<dependency>`
`<groupId>com.github.NuVotifier.NuVotifier</groupId>`
`<artifactId>nuvotifier-bukkit</artifactId>`
` <version>2.6.0</version>`
` <scope>provided</scope>`
`</dependency>`

This fixed it for me.

**Comment by maxnitschke** [comment]
@Ichbinjoe After 3 hours of research and trying plenty of the aforementioned workarounds, I can't manage to get the following imports resolved properly (especially the first one):

```
import com.vexsoftware.votifier.bungee.events.VotifierEvent;
import com.vexsoftware.votifier.model.Vote;
```
Which maven repository and dependency do I need to use in order to get my vote listener working on bungeecord?

The following configuration (as mentioned in your developer guide) seems to not work anymore (Failure to find com.vexsoftware:nuvotifier-universal:pom:2.6.0):

```
<project>
...
    <repositories>
        <repository>
            <id>jitpack</id>
            <url>https://jitpack.io</url>
        </repository>
        ...
    </repositories>
...
    <dependencies>
        <dependency>
            <groupId>com.vexsoftware</groupId>
            <artifactId>nuvotifier-universal</artifactId>
            <version>2.6.0</version>
            <scope>provided</scope>
        </dependency>
        ...
    </dependencies>
...
</project>
```

**Comment by socketbat** [comment]
> @Ichbinjoe After 3 hours of research and trying plenty of the aforementioned workarounds, I can't manage to get the following imports resolved properly (especially the first one):
> 
> 
> 
> ```
> 
> import com.vexsoftware.votifier.bungee.events.VotifierEvent;
> 
> import com.vexsoftware.votifier.model.Vote;
> 
> ```
> 
> Which maven repository and dependency do I need to use in order to get my vote listener working on bungeecord?
> 
> 
> 
> The following configuration (as mentioned in your developer guide) seems to not work anymore (Failure to find com.vexsoftware:nuvotifier-universal:pom:2.6.0):
> 
> 
> 
> ```
> 
> <project>
> 
> ...
> 
>     <repositories>
> 
>         <repository>
> 
>             <id>jitpack</id>
> 
>             <url>https://jitpack.io</url>
> 
>         </repository>
> 
>         ...
> 
>     </repositories>
> 
> ...
> 
>     <dependencies>
> 
>         <dependency>
> 
>             <groupId>com.vexsoftware</groupId>
> 
>             <artifactId>nuvotifier-universal</artifactId>
> 
>             <version>2.6.0</version>
> 
>             <scope>provided</scope>
> 
>         </dependency>
> 
>         ...
> 
>     </dependencies>
> 
> ...
> 
> </project>
> 
> ```

I had the same problem. It's frustrating but I ended up using the bukkit version and used redis pub/sub to send the votes to all servers.

**Comment by maxnitschke** [comment]
> I had the same problem. It's frustrating but I ended up using the bukkit version and used redis pub/sub to send the votes to all servers.

Since this issue hasn't been resolved for more than 4 months now I can see where you're coming from. I'd really appreciate if we could just have a public repository for the nuvotifier-universal version though. Especially given the fact that we already have one for the bukkit version. @Ichbinjoe Would it be possible to provide the nuvotifier-universal version at one of your repositories?

Upon further investigating this issue, I actually managed to find a public repository that provides the universal jar: [https://repo.acrylicstyle.xyz/com/vexsoftware/nuvotifier-universal/2.6.0/](https://repo.acrylicstyle.xyz/com/vexsoftware/nuvotifier-universal/2.6.0/). However, since this is not the official repository: Use at your own risk.

**Comment by milkyway0308** [comment]
I noticed that NuVotifier is not available on JitPack, and I have decided to host the assets on my own repository for a while.
If anyone requires access, feel free to use my repository. 
You can find it at the following address: https://repo.trinarywolf.net/public
